components/script/dom/webidls/DOMTokenList.webidl

```diff
@@ -21,9 +21,8 @@ interface DOMTokenList {
   [CEReactions, Throws]
   void replace(DOMString token, DOMString newToken);
 
-  [CEReactions, Pure]
-           attribute DOMString value;
+  [CEReactions]
```

Comment:
Removed `Pure` because WebIDL.py currently does not allow it on stringifier attribute (just because gecko didn't have it on any of its stringifier attributes).

The support can be added, but not sure why servo has Pure but gecko does not. @nox, could you give me some idea?

Reply:
Thanks for the docs, but I mean why DOMTokenList in Gecko does not have Pure.

Reply:
In other words, should we upstream the changes in #7601 ?

Reply:
No clue, I don't know if that conflicts with how stringifiers are handled through SpiderMonkey. Or it could be a case of no one adding support for `[Pure]` on them in the Gecko side.

Reply:
There is no blocker to add `[Pure]` support on stringifier attributes, it's just that no one was bothered to add one on DOMTokenList or to do what #7601 did.

@bzbarsky, thoughts?

Reply:
I'm not actually confident that Pure has any meaning in Servo's implementation. It may have been copied from upstream webidl in Gecko.

Reply:
> I'm not actually confident that Pure has any meaning in Servo's implementation. It may have been copied from upstream webidl in Gecko.

`WebIDL.py` translates `[Pure]` and `[Constant]` to specific `[Affects]`/`[DependsOn]` combinations, and you can actually find references to those in `CodegenRust.py`.

Reply:
(Submitted too fast.)

And in #7601 I went out of my way to make use of them more extensively so those weren't there by coincidence.

Reply:
Fwiw, I think we should add support for [Pure] in the parser here and probably add it in the Gecko IDL.